### PR TITLE
add log forwarding and fetch-lib

### LINKED ISSUE #14: Grafana panel doesn't show the instance addresses in `SSL Probes: Instance whose certificate is about to expire`
### Bug Description

image

The bottom panel `Expired in less than X days` doesn't show the instance addresses like URL but only shows the identical juju_model_uuid + juju_application for multiple items.

```
probe_ssl_earliest_cert_expiry{juju_application=~"$juju_application",juju_model=~"$juju_model",juju_model_uuid=~"$juju_model_uuid",juju_unit=~"$juju_unit",probe_target=~"$probe_target"} - time() < (3600 * 24 * 30)
```

### To Reproduce

```
$ juju deploy blackbox-exporter-k8s blackbox-exporter --channel latest/edge

$ juju config blackbox-exporter probes_file="
scrape_configs:
  - job_name: 'blackbox_k8s_endpoints'
    metrics_path: /probe
    params:
      module: [http_2xx]
    static_configs:
      - targets:
        - http://ubuntu.com
        - https://ubuntu.com
        - https://releases.ubuntu.com
"

$ juju relate prometheus:metrics-endpoint blackbox-exporter:self-metrics-endpoint
$ juju relate grafana:grafana-dashboard blackbox-exporter:grafana-dashboard
```

Then open Grafana, and change the query to `< (3600 * 24 * 90)` to fill the panel just for testing.

### Environment

blackbox-exporter-k8s         latest/edge    3

### Relevant log output

```shell
N/A
```

### Additional context

image

### LINKED ISSUE #16: No alert even when certificate is about to expire nor a target is down
### Bug Description

There is no alert defined by the charm for some important or critical events.
- certification is about to expire (e.g. warning when < 30 days, critical when < 7 days)
- a target is down

Alerts are always happy even with the events above.

image

image

For the certificate expiration part, we need something like this:
https://grafana.com/blog/2020/11/25/how-we-eliminated-service-outages-from-certificate-expired-by-setting-up-alerts-with-grafana-and-prometheus/#alerting

### To Reproduce
```
$ juju deploy blackbox-exporter-k8s blackbox-exporter --channel latest/edge

$ juju config blackbox-exporter probes_file="
scrape_configs:
  - job_name: 'blackbox_k8s_endpoints'
    metrics_path: /probe
    params:
      module: [http_2xx]
    static_configs:
      - targets:
        - http://ubuntu.com
        - https://ubuntu.com
        - https://releases.ubuntu.com
        - https://non-existent.canonical.com/
"

$ juju relate prometheus:metrics-endpoint blackbox-exporter:self-metrics-endpoint
$ juju relate grafana:grafana-dashboard blackbox-exporter:grafana-dashboard
```

### Environment

blackbox-exporter-k8s latest/edge 3

### Relevant log output

```shell
N/A
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lucabello**
This PR does a number of things:

* Closes #14 
* Relates to #15
* Closes #16
* Implement Pebble Log Forwarding
* `charmcraft fetch-lib`

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -107,13 +107,7 @@ def __init__(self, *args):
...
         )
         self._grafana_dashboard_provider = GrafanaDashboardProvider(charm=self)
-        self._log_proxy = LogProxyConsumer(
-            charm=self,
-            relation_name="logging",
-            log_files=[self._log_path],
-            container_name=self._container_name,
-            enable_syslog=False,
-        )
+        self._log_forwarding = LogForwarder(self, relation_name="logging")
```
Comment (sed-i) [inline src/charm.py:110]: Does this mean we need to `assumes juju >= 3.4`?

Comment (lucabello) [reply src/charm.py:110]: Done!

**Review by PietroPasotti — CHANGES_REQUESTED** [review]
this PR does too many things at once. 
Can we break it up?

**Comment by lucabello** [comment]
The bulk of the changed files is due to `charmcraft fetch-lib`. The actual changes are one line in the charm (to instantiate `LogForwarder`), a dashboard change, and a new alert rule file.

I think that's fine, but if you'd rather split it into 4 PRs I can go ahead and do that!

**Review by PietroPasotti — APPROVED** [review]
lgtm

**Review by PietroPasotti — APPROVED** [review]
lgtm

**Pull request merged**